std/traits.d

```diff
@@ -5998,6 +6000,62 @@ enum bool isNarrowString(T) = (is(T : const char[]) || is(T : const wchar[])) &&
     }
 }
 
+/**
+ * Detects whether `T` is a comparable type. Basic types and structs and
+ * classes that implement opCmp are ordering comparable. All user defined
+ * types are equality comparable.
+ */
+enum bool isOrderingComparable(T) = is(typeof(ifTestable!(T, (T t) => t < t))) &&
+                                    is(typeof((T t) => t < t));
```

Comment:
I think I figure the problem you encountered, but it's incorrect because you still need to make sure the type returned by `t < t` is if-testable. A simple solution is:

```d
enum bool isOrderingComparable(T) = ifTestable!(T, unaryFun!"a < a");
```

Due to the way `unaryFun` works, you're protected against errors if the type does not define opCmp.